### DISCUSSION TITLE
auth hook unexpected end of json

### DISCUSSION THREAD

**Discussion by Mohhaliim** [supabase/supabase#22162] [Questions]
as the title says I've created a hook to run on login and attach more data to the claims following the docs
`declare
    claims jsonb;
    user_data jsonb;
  begin
    
    -- Query the users table in the public schema
    select jsonb_build_object('role', u.role, 'name', u.name, 'email', u.email, 'code', u.code) into user_data
    from public.users u
    where u.id = (event->>'user_id')::uuid;

    -- Attach the user data to the claims
    claims := event->'claims';
    claims := jsonb_set(claims, '{user_data}', user_data);

    -- Return the modified event
    event := jsonb_set(event, '{claims}', claims);

    return event;
  end;`

when I check the auth logs this has an error of "unexpected end of json" i tried to hardcode the id but it's not the problem tried to run this query in the SQL editor and it worked pretty fine and created a table with the jsob obj in a column, what might be my issue here?

**Comment by silentworks** [comment]
This looks fine to me. I'd start commenting out lines in the code to see where exactly the error is happening.

**Reply by Mohhaliim** [reply]
issue was that the hook doesn't have access to the public.users table and is returning null, fixed by setting rls to anon on users table, but I don't like this fix tbh, I tried to grant it access to public schema but still doesn't work
